fd/diff.go

```diff
@@ -57,23 +60,30 @@ func Derivative(f func(float64) float64, x float64, settings *Settings) float64
 	if settings == nil {
 		settings = DefaultSettings()
 	}
-	step := settings.Method.Step
+	formula := settings.Formula
+	if formula.Stencil == nil {
+		formula = Central
+	}
+	step := settings.Step
+	if step == 0 {
+		step = formula.Step
+	}
+
 	var deriv float64
-	method := settings.Method
-	if !settings.Concurrent {
-		for _, pt := range method.Stencil {
+	if settings.Concurrent <= 0 {
+		for _, pt := range formula.Stencil {
 			if settings.OriginKnown && pt.Loc == 0 {
 				deriv += pt.Coeff * settings.OriginValue
 				continue
 			}
 			deriv += pt.Coeff * f(x+step*pt.Loc)
 		}
-		return deriv / math.Pow(step, float64(method.Order))
+		return deriv / math.Pow(step, float64(formula.Order))
 	}
 
 	wg := &sync.WaitGroup{}
 	mux := &sync.Mutex{}
-	for _, pt := range method.Stencil {
+	for _, pt := range formula.Stencil {
```

Comment:
Yea, perhaps. This didn't originally because these days GOMAXPROCS is usually at least 4, typically 8 or more. Finite difference stencils of significantly larger that 8 points are not used very often. It should probably be that if len(stencil) < concurrent, it uses this algorithm, otherwise it launches workers in the typical pattern.